featurebyte/service/item_table.py

```diff
@@ -22,21 +30,69 @@ class ItemTableService(
     document_class = ItemTableModel
     document_update_class = ItemTableServiceUpdate
 
+    def __init__(
+        self,
+        user: Any,
+        persistent: Persistent,
+        catalog_id: Optional[ObjectId],
+        block_modification_handler: BlockModificationHandler,
+        feature_store_service: FeatureStoreService,
+        storage: Storage,
+        redis: Redis[Any],
+        event_table_service: EventTableService,
+    ):
+        super().__init__(
+            user=user,
+            persistent=persistent,
+            catalog_id=catalog_id,
+            block_modification_handler=block_modification_handler,
+            feature_store_service=feature_store_service,
+            storage=storage,
+            redis=redis,
+        )
+        self.event_table_service = event_table_service
+
     @property
     def class_name(self) -> str:
         return "ItemTable"
 
+    async def create_document(self, data: ItemTableCreate) -> ItemTableModel:
+        # validate event ID column in the event table has matching data type with that in the item table
+        item_table_event_id_dtype = [
+            column_info
+            for column_info in data.columns_info
+            if column_info.name == data.event_id_column
+        ][0].dtype
```

Comment:
If i understood this correctly, when the code execution reaches here, we already validate the `event_id_column` must be one of the column in the `columns_info`. https://github.com/featurebyte/featurebyte/blob/c2f570fe2eca09ba91ea00c840a60892101e6fc2/featurebyte/api/item_table.py#L115

Reply:
yes there is a special column validator in the `ItemTableCreate` pydantic model and `event_id_column` is a non-optional column so the schema validation should fail before this if that is not specified, or if the column does not exist in the table